src/omv/imlib/framebuffer.c

```diff
@@ -237,14 +282,46 @@ uint32_t framebuffer_get_buffer_size()
     return (size / 32) * 32;
 }
 
-uint8_t *framebuffer_get_buffer()
+framebuffer_t *framebuffer_get_framebuffer()
 {
-    return framebuffer->pixels;
+   return framebuffer;
 }
 
-void framebuffer_set(int32_t w, int32_t h, int32_t bpp)
+void framebuffer_swap_buffers(uint32_t offset)
 {
-    framebuffer->w = w;
-    framebuffer->h = h;
-    framebuffer->bpp = bpp;
+    // Save the current frame buffer before swapping.
+    framebuffer_t *old_fb = framebuffer;
+
+    // Set/Clear frame buffer flags.
+    FB_CLR_FLAG(old_fb, FB_USED);
+    FB_SET_FLAG(old_fb, FB_DIRTY);
+
+    // Swap the main frame buffer pointer.
+    for (int i=0; i<N_BUFFERS; i++) {
+        if (framebuffer != &buffers[i]) {
+            framebuffer = &buffers[i];
+            break;
+        }
+    }
+
+    // Initialize from the first buffer, copying x,y,w,h,u,v, etc...
+    framebuffer->x = old_fb->x;
+    framebuffer->y = old_fb->y;
+    framebuffer->w = old_fb->w;
+    framebuffer->h = old_fb->h;
+    framebuffer->u = old_fb->u;
+    framebuffer->v = old_fb->v;
+    framebuffer->x = old_fb->x;
+    framebuffer->bpp  = -1;
+
+    // The second buffer needs to be initialized from fb0
+    if (framebuffer != &buffers[0] && framebuffer->pixels == &_fb_base) {
+        // If this is first time this buffer is used, adjust its pixels pointer
+        // offset within the raw buffer to point just after the first frame buffer.
+        framebuffer->pixels = &_fb_base + FB_ALIGN_SIZE(offset);
```

Comment:
Users are allowed to change the pixformat/framesize while a script is running. I don't see this recomputed ever again. You need a way to reset everything when a user changes any major setting.

Reply:
Yes, we need to reset buffers just like MAIN_FB()->bpp = -1;